operator-wg/whitepaper-community-review/README.md

```diff
@@ -0,0 +1,875 @@
+
+The operator design pattern defines how to manage application and infrastructure resources using declarative domain-specific knowledge. The goal of the pattern is to reduce the amount of manual imperative work which is required to keep an application in a healthy and well-maintained state.
+
+By using the operator pattern, the knowledge on how to adjust and maintain a resource is captured in code and usually in a single service (also called a controller).
+
+When using an operator design pattern the user should only be required to describe the desired state of the application and resources. A software pre-written should make the necessary changes in the world so it will be in the desired state. The software will also monitor the real state continuously and take actions to keep it healthy and n the same state (preventing drifts)
+
+A general diagram of an operator will have software that can read the desired spec and can create and manage the resources that were described. 
+
+![Operator Design Pattern](img/02_1_operator_pattern.png)
+
+The Operator pattern consists of three components:
+
+* The application or infrastructure that we want to manage in a declarative way.
+* Our domain specific knowledge that describes the state of the application in a declarative way.
+* A controller that run continuously:
+  * Reads and is aware of the state.
+  * Runs actions against the application in an automated way.
+  * Report the state of the application  in a declarative way.
+
+This design pattern will be applied on Kubernetes and its operators in the next sections.
+
+### Operator Characteristics
+The core purpose of any operator is to extend its orchestrator's underlying API with new domain knowledge. As an example, an orchestration platform as Kubernetes natively understands things like containers and layer 4 load balancers via the Pod and Service objects. An operator adds new capabilities for more complex systems and applications. For instance, a prometheus-operator introduces new object types, "Prometheus", which extend Kubernetes with high-level support for deploying and running Prometheus servers.
+
+The capabilities provided by an operator can be sorted into three overarching categories: dynamic configuration, operational automation, and domain knowledge.
+
+#### Dynamic Configuration
+Since the dawn of time, there have been two main ways to configure software, configuration files and environment variables. In the cloud native world there are some newer options like querying a well-known API at startup, but most existing software in the world uses one or both of these options. Kubernetes naturally provides many tools to interact with these (ConfigMaps, Secrets, etc) but because they are generic, they don’t understand any specifics of configuring a given application. An operator can define new custom object types (custom resources) to better express the configuration of a particular application in a Kubernetes context.
```

Comment:
```suggestion
Since the dawn of time, there have been two main ways to configure software, configuration files and environment variables. In the cloud-native world, there are some newer options like querying a well-known API at startup, but most existing software in the world uses one or both of these options. Kubernetes naturally provides many tools to interact with these (such as ConfigMaps and Secrets) but because they are generic, they don’t understand any specifics of configuring a given application. An operator can define new custom object types (custom resources) to better express the configuration of a particular application in a Kubernetes context.
```